src/details/ArborX_DetailsMortonCode.hpp

```diff
@@ -23,9 +24,12 @@ namespace Details
 // Magic numbers generated by the script in
 // https://stackoverflow.com/questions/1024754/how-to-compute-a-3d-morton-number-interleave-the-bits-of-3-ints/18528775#18528775
 
+template <int DIM>
+KOKKOS_INLINE_FUNCTION unsigned int expandBitsBy(unsigned int);
```

Comment:
Maybe go for a failed static assertion rather than a link-time error for `DIM` that do not have an implementation

Reply:
Not trivial (see [here](https://stackoverflow.com/questions/14637356/static-assert-fails-compilation-even-though-template-function-is-called-nowhere)), but did it.

Reply:
yes trivial, `static_assert(0 < DIM && DIM < 6)`

Reply:
Why am i so stupid... Sigh.